catalog/rest/rest.go

```diff
@@ -276,6 +280,10 @@ func doDelete[T any](ctx context.Context, baseURI *url.URL, path []string, cl *h
 	return do[T](ctx, http.MethodDelete, baseURI, path, cl, override, true)
 }
 
+func doHead[T any](ctx context.Context, baseURI *url.URL, path []string, cl *http.Client, override map[int]error) (ret T, err error) {
+	return do[T](ctx, http.MethodHead, baseURI, path, cl, override, true)
+}
```

Comment:
Since a `HEAD` request doesn't have any result/output, shouldn't we just remove the template param and the `ret` result?